fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -581,28 +598,198 @@ void basicGroupedPartitionedTest() {
+            long start = Instant.now().toEpochMilli();
+            for (int i = 0; i < totalDocCount; i++) {
+                recordStore.saveRecord(createComplexDocument(1000L + i, ENGINEER_JOKE, 1, start + i * 100));
+            }
+            commit(context);
+        }
+
+        // initially, all documents are saved into one partition
+        List<LucenePartitionInfoProto.LucenePartitionInfo> partitionInfos = getPartitionMeta(COMPLEX_PARTITIONED,
+                Tuple.from(1), contextProps, schemaSetup);
+        assertEquals(1, partitionInfos.size());
+        assertEquals(totalDocCount, partitionInfos.get(0).getCount());
+
+        // run re-partitioning
+        explicitMergeIndex(COMPLEX_PARTITIONED, contextProps, schemaSetup);
+        partitionInfos = getPartitionMeta(COMPLEX_PARTITIONED,
+                Tuple.from(1), contextProps, schemaSetup);
+        // now there should be 2 partitions: partition 0 with (totalDocCount - 6) docs, and partition 1 with 6 docs
+        assertEquals(2, partitionInfos.size());
+        assertEquals(0, partitionInfos.get(0).getId());
+        assertEquals(totalDocCount - 6, partitionInfos.get(0).getCount());
+        assertEquals(1, partitionInfos.get(1).getId());
+        assertEquals(6, partitionInfos.get(1).getCount());
+
+        // run re-partitioning again
+        explicitMergeIndex(COMPLEX_PARTITIONED, contextProps, schemaSetup);
+        partitionInfos = getPartitionMeta(COMPLEX_PARTITIONED,
+                Tuple.from(1), contextProps, schemaSetup);
+        // now there should be 3 partitions: partition 0 with (totalDocCount - 12) docs, partition 2 with 6 docs,
+        // and partition 1 with 6 docs
+        assertEquals(3, partitionInfos.size());
+        assertEquals(0, partitionInfos.get(0).getId());
+        assertEquals(totalDocCount - 12, partitionInfos.get(0).getCount());
+        assertEquals(2, partitionInfos.get(1).getId());
+        assertEquals(6, partitionInfos.get(1).getCount());
+        assertEquals(1, partitionInfos.get(2).getId());
+        assertEquals(6, partitionInfos.get(2).getCount());
+
+        // partition metadata validated above. now validate that the documents have indeed been
+        // moved.
+        // expected: docs 1000-1005 -> partition 1 (oldest)
+        //                1006-1011 -> partition 2 (middle)
+        //                1012-1019 -> partition 0 (newest)
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+
+            byte[] continuation = validateLastDocumentInPartition(null, 0, 7, Tuple.from(1L, 1012));
+            continuation = validateLastDocumentInPartition(continuation, 2, 5, Tuple.from(1L, 1006));
+            validateLastDocumentInPartition(continuation, 1, 5, Tuple.from(1L, 1000));
+        }
+    }
+
+    private byte[] validateLastDocumentInPartition(byte[] continuation, int expectedPartitionId, int amountToSkip, Tuple expectedDocId)
```

Comment:
I wonder if it would be worthwhile to have a search that explicitly fetches the `IndexWriter` from the group/partition and makes sure that searching for everything in there finds the set of documents that should be there.

Reply:
you mean `IndexReader`?